src/processor.rs

```diff
@@ -1,18 +1,171 @@
+            return Err(ProgramError::InvalidAccountOwner);
+        }
+
+        let mint_data = mint_info.try_borrow_data()?;
+        let mint = StateWithExtensions::<Mint>::unpack(&mint_data)?;
+
+        // Ensure the mint is configured with the `TransferHook` extension,
+        // and the program ID is the Paladin Rewards program.
+        let transfer_hook = mint.get_extension::<TransferHook>()?;
+        let hook_program_id: Option<Pubkey> = transfer_hook.program_id.into();
+        if !hook_program_id.eq(&Some(*program_id)) {
+            return Err(PaladinRewardsError::IncorrectTransferHookProgramId.into());
+        }
+
+        // Ensure the provided mint authority is the correct mint authority.
+        if !mint
+            .base
+            .mint_authority
+            .eq(&COption::Some(*mint_authority_info.key))
+        {
+            return Err(PaladinRewardsError::IncorrectMintAuthority.into());
+        }
+
+        // Ensure the mint authority is a signer.
+        if !mint_authority_info.is_signer {
+            return Err(ProgramError::MissingRequiredSignature);
+        }
+    }
+
+    // Initialize the holder rewards pool account.
+    {
+        let (holder_rewards_pool_address, holder_rewards_pool_bump) =
+            get_holder_rewards_pool_address_and_bump_seed(mint_info.key);
+        let holder_rewards_pool_signer_seeds = &[
+            SEED_PREFIX_HOLDER_REWARDS_POOL,
+            mint_info.key.as_ref(),
+            &[holder_rewards_pool_bump],
+        ];
+
+        // Ensure the provided holder rewards pool address is the correct
+        // address derived from the mint.
+        if !holder_rewards_pool_info
+            .key
+            .eq(&holder_rewards_pool_address)
+        {
+            return Err(PaladinRewardsError::IncorrectHolderRewardsPoolAddress.into());
+        }
+
+        // Ensure the holder rewards pool account has not already been
+        // initialized.
+        if holder_rewards_pool_info.data_len() != 0 {
+            return Err(ProgramError::AccountAlreadyInitialized);
+        }
+
+        // Allocate & assign.
+        invoke_signed(
+            &system_instruction::allocate(
+                &holder_rewards_pool_address,
+                std::mem::size_of::<HolderRewardsPool>() as u64,
+            ),
+            &[holder_rewards_pool_info.clone()],
+            &[holder_rewards_pool_signer_seeds],
+        )?;
+        invoke_signed(
+            &system_instruction::assign(&holder_rewards_pool_address, program_id),
+            &[holder_rewards_pool_info.clone()],
+            &[holder_rewards_pool_signer_seeds],
+        )?;
+
+        // Write the data.
+        let holder_rewards_pool_state = HolderRewardsPool::default();
+        let holder_rewards_pool_data = bytemuck::bytes_of(&holder_rewards_pool_state);
+        holder_rewards_pool_info
+            .try_borrow_mut_data()?
+            .copy_from_slice(holder_rewards_pool_data);
```

Comment:
nit: Rather than instantiating a standalone `HolderRewardsPool` and then copying its data into the account, it might be better to simply instantiate the account data, ie:

```suggestion
        // Write the data.
        let data = holder_rewards_pool_info.try_borrow_mut_data()?;
        *bytemuck::try_from_bytes_mut(data)? = HolderRewardsPool::default();
```